### PR TITLE
[Merged by Bors] - feat: normed structure instances for SetLike subobjects

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Normed/Field/Basic.lean [hunk 1]
```diff
@@ -1301,3 +1339,4 @@ end SubringClass
 
 -- Guard against import creep.
 assert_not_exists RestrictScalars
+#lint
```
Comment (j-loreaux) [inline Mathlib/Analysis/Normed/Field/Basic.lean:1342]:
```suggestion
```

File: Mathlib/Analysis/Normed/Field/Basic.lean [hunk 2]
```diff
@@ -280,13 +280,33 @@ instance NonUnitalSubalgebra.nonUnitalSeminormedRing {𝕜 : Type*} [CommRing 
...
+
 /-- A non-unital subalgebra of a non-unital normed ring is also a non-unital normed ring, with the
 restriction of the norm.  -/
 instance NonUnitalSubalgebra.nonUnitalNormedRing {𝕜 : Type*} [CommRing 𝕜] {E : Type*}
     [NonUnitalNormedRing E] [Module 𝕜 E] (s : NonUnitalSubalgebra 𝕜 E) : NonUnitalNormedRing s :=
   { s.nonUnitalSeminormedRing with
     eq_of_dist_eq_zero := eq_of_dist_eq_zero }
 
+/-- A non-unital subalgebra of a non-unital seminormed ring is also a non-unital seminormed ring,
+with the restriction of the norm.  -/
```
Comment (dupuisf) [inline Mathlib/Analysis/Normed/Field/Basic.lean:302]:
```suggestion
/-- A non-unital subalgebra of a non-unital normed ring is also a non-unital normed ring,
with the restriction of the norm.  -/
```

File: Mathlib/Analysis/Normed/Field/Basic.lean [hunk 3]
```diff
@@ -347,13 +367,31 @@ instance Subalgebra.seminormedRing {𝕜 : Type*} [CommRing 𝕜] {E : Type*} [S
...
+
 /-- A subalgebra of a normed ring is also a normed ring, with the restriction of the norm. -/
 instance Subalgebra.normedRing {𝕜 : Type*} [CommRing 𝕜] {E : Type*} [NormedRing E]
     [Algebra 𝕜 E] (s : Subalgebra 𝕜 E) : NormedRing s :=
   { s.seminormedRing with
     eq_of_dist_eq_zero := eq_of_dist_eq_zero }
 #align subalgebra.normed_ring Subalgebra.normedRing
 
+/-- A subalgebra of a seminormed ring is also a seminormed ring, with the restriction of the
+norm. -/
```
Comment (dupuisf) [inline Mathlib/Analysis/Normed/Field/Basic.lean:388]:
```suggestion
/-- A subalgebra of a normed ring is also a normed ring, with the restriction of the
norm. -/
```

**Review by dupuisf — COMMENTED** [review]
Looks good!

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**